compiler/rustc_target/src/asm/mod.rs

```diff
@@ -309,6 +313,7 @@ impl InlineAsmReg {
             Self::RiscV(r) => r.emit(out, arch, modifier),
             Self::Hexagon(r) => r.emit(out, arch, modifier),
             Self::Mips(r) => r.emit(out, arch, modifier),
+            Self::Err => unreachable!(),
```

Comment:
Many of these `unreachable!`s here and in `{modifier,reg}_to_llvm` feel like they should be a `bug!` instead. These aren't very obviously unreachable and in the future it might be the case that somebody does change some code in a way that would cause this to become reached too (thus introducing a bug).

Reply:
`bug!` is defined in `rustc_middle` which `rustc_target` can't depend on since that would result in a circular dependency. I've added a message to the `unreachable` though.